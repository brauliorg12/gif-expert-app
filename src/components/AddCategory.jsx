import { useState } from 'react';

export const AddCategory = ({ onNewCategoryFn }) => {
  const [inputValue, setInputValue] = useState('');

  // Funcion que se ejecuta al cambiar el valor del input
  // Se utiliza para actualizar el estado del input
  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmitFn = (event) => {
    // Evitar el comportamiento por defecto del formulario
    event.preventDefault();
    // Evitar enviar menos de 1 caracter
    if (inputValue.trim().length <= 1) return;

    // Limpiar el input
    setInputValue('');

    // Emitir el evento onNewCategory
    onNewCategoryFn(inputValue.trim());
  };

  return (
    <form onSubmit={onSubmitFn} aria-label="form">
      {/* Input */}
      <input
        type="text"
        placeholder="Buscar gifs..."
        value={inputValue}
        onChange={(event) => onInputChange(event)}
      />
    </form>
  );
};
