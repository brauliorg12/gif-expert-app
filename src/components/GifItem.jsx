import PropTypes from 'prop-types';

export const GifItem = ({ url, title }) => {
  return (
    <>
      <li className="card">
        <img src={url} alt={title} />
        <a>{title}</a>
      </li>
    </>
  );
};

GifItem.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};
