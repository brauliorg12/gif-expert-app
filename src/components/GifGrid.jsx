import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {isLoading && <p>Cargando...</p>}

      <ol className="card-grid">
        {images?.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </ol>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
