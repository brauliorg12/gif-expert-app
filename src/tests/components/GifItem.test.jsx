/* eslint-disable no-undef */
import { render } from '@testing-library/react';
import { GifItem } from '../../components/GifItem';

describe('Pruebas en <GifItem />', () => {
  test('debe de hacer match con el snapshot', () => {
    const title = 'Example Title';
    const url = 'https://example.com/image.jpg';

    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
    const title = 'Example Title';
    const url = 'https://example.com/image.jpg';

    const { getByRole } = render(<GifItem title={title} url={url} />);
    const { src, alt } = getByRole('img');

    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('debe de mostrar el tÃ­tulo en el componente', () => {
    const title = 'Example Title';
    const url = 'https://example.com/image.jpg';

    const { getByText } = render(<GifItem title={title} url={url} />);
    expect(getByText(title)).toBeTruthy();
  });
});
