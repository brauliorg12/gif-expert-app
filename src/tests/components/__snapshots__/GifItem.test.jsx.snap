// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe de hacer match con el snapshot 1`] = `
<div>
  <li
    class="card"
  >
    <img
      alt="Example Title"
      src="https://example.com/image.jpg"
    />
    <a>
      Example Title
    </a>
  </li>
</div>
`;
